### PR TITLE
Fix bug that causes videos to keep playing audio but freeze the video on iOS

### PR DESCRIPTION
**Pull request by aalzanki**
Test Plan:
 - the easiet way to test this is to go to the KA app and make this change locally in xcode.
 - Run the app on ios
 - go to a bug of videos
 - keep toggling the speed preference. Keeps doing this over and over again. Before this fix, the video would freeze. After this fix, the video doesn't freeze

https://khanacademy.atlassian.net/browse/MOB-2602

### CODE REVIEW

**Comment by aalzanki** [comment]
Looking at https://developer.apple.com/documentation/avfoundation/avplayer/1386726-play 

It doesn't really make sense to be playing the video and then setting the rate. Setting the rate is actually enough to play and pause videos on iOS, so I'm not really sure why the module isn't just doing that. The exact cause of the bug isn't clear to me because it seems like it probably has to do with the implementation details of iOS.

**Comment by aalzanki** [comment]
I was hoping to make a pull request to the RN video repo, but unfortunately we're way behind and it's not trivial to bring us to the latest version, specifically on Android.

File: ios/Video/RCTVideo.m [hunk 1]
```diff
@@ -828,15 +828,13 @@ - (void)setIgnoreSilentSwitch:(NSString *)ignoreSilentSwitch
...
   if (paused) {
-    [_player pause];
     [_player setRate:0.0];
   } else {
     if([_ignoreSilentSwitch isEqualToString:@"ignore"]) {
       [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback error:nil];
     } else if([_ignoreSilentSwitch isEqualToString:@"obey"]) {
       [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient error:nil];
     }
-    [_player play];
```
Comment (bryanjclark) [inline ios/Video/RCTVideo.m:839]: This pause-then-play-again thing is... interesting. Looks like it was introduced in some super-poorly-documented commit back in 2015:
https://github.com/react-native-community/react-native-video/commit/f2b0dbd2db81f2e03f2cee457eb4e465ec111b76

...so yeah, sounds good to me to remove this hacky bit!

**Review by bryanjclark — APPROVED** [review]

**Review by jeremywiebe — APPROVED** [review]
Makes sense.  👍

**Pull request merged**